pkg/cache/clusterqueue_test.go

```diff
@@ -646,6 +653,29 @@ func TestClusterQueueUpdateWithAdmissionCheck(t *testing.T) {
 			wantStatus: pending,
 			wantReason: "MultipleSingleInstanceControllerChecks",
 		},
+		{
+			name:     "Active clusterQueue with an AC per ResourceFlavor",
+			cq:       cqWithACPerFlavor,
+			cqStatus: pending,
+			admissionChecks: map[string]AdmissionCheck{
+				"check1": {
+					Active:                true,
+					Controller:            "controller1",
+					ApplyOnlyToAllFlavors: true,
+				},
+				"check2": {
+					Active:     true,
+					Controller: "controller2",
+				},
+				"check3": {
+					Active:                true,
+					Controller:            "controller3",
+					ApplyOnlyToAllFlavors: true,
+				},
+			},
+			wantStatus: pending,
+			wantReason: "AllFlavorsChecksAppliedToSpecificFlavor",
```

Comment:
It would be nice to describe more details in the message, for example "The Admission Check {name} (controlled by {controller}) can only be applied to all flavors, but applied to {list of flavors}.".

Reply:
Do you mean that we should include the details in the reason string? As in the other cases of  [`inactiveReason()`](https://github.com/kubernetes-sigs/kueue/pull/2047/files#diff-441da8393964615cf5b020d0afb9199588e157e50718e83b527788eb679391ecR356) method, currently we're returning just the reason without extra information.

Reply:
> Do you mean that we should include the details in the reason string?

The `reason` field is ok, this is just a camel-case summary. 

I was thinking about the `message` field that is set on the ClusterQueue's condition when `Active=False`. 
This indeed is returned as the second param from `inactiveReason`, and computed here as a concatenation of `reason` fields. However, it does not tell the user details, like which AC we are referring to.

To pass this information we would need to probably compute it inside `updateWithAdmissionChecks` and store [here](https://github.com/kubernetes-sigs/kueue/blob/main/pkg/cache/clusterqueue.go#L87-L91), probably as a map from reason to message.

Since this is a pre-existing limitation of the current structure of the code I'm ok to leave it as a follow up. WDYT?

Reply:
I agree, I think we can leave it as a follow up and include there the details for this and the rest of the admission checks as well.

Reply:
sgtm